pkg/cluster_agent/events_manager.py

```diff
@@ -37,18 +36,46 @@ async def get_event(self) -> KubernetesEvent:
         """
         raise NotImplementedError
 
-
-    async def get_events(self, max_size: int) -> List[KubernetesEvent]:
+    async def _read_event(self, timeout: int=None):
+        """
+        Reads and returns an event. If timeout is given, then trying to read event up to
+        the timeout given value.
+        In case of timeout, returns None.
+        """
+        event = None
+        if not timeout:
+            event = await self.get_event()
```

Comment:
The timeout param in wait_for can be None and then it acts like a regular await, I think we can remove the check if there is timeout and always call `await wait_for(self.get_event(), timeout=timeout)`